client/my-sites/stats/overview.jsx

```diff
@@ -82,9 +83,16 @@ class StatsOverview extends Component {
 			);
 		} );
 
+		const date = moment()
+			.utcOffset( 0 ) // Don't have one site to choose a timezone from...
+			//.utcOffset( Number.isFinite( gmtOffset ) ? gmtOffset : 0 )
+			.format( 'YYYY-MM-DD' );
+		const query = { date, period }; // Makes a new object each render = bad
```

Comment:
Yeah, the real issue here is `QuerySiteStats` taking its query as an object. That API shape isn't great for a component, precisely because it lends itself to this sort of deopt.

However, the component does have `componentDidUpdate` method that I added years ago to mitigate the issue, which does a shallow comparison on `query`. So, given that both `date` and `period` are primitives, we should be okay.

Reply:
Oh, nice! I tested it out and it's working perfectly.